CONTRIBUTING.md

```diff
@@ -159,17 +167,47 @@ project:
 8. [Open a pull request](https://help.github.com/articles/using-pull-requests/)
     with a clear title and description.
 
-**IMPORTANT**: By submitting a pull request, you agree to allow the project owner to
-license your work under the same license as that used by the project.
+**IMPORTANT**: By submitting a pull request, you agree to allow the project
+owner to license your work under the same license as that used by the project,
+see also [Legal Stuff](#legal).
 
 <a name="branching"></a>
 ### Branching
 
-`igraph` is committed to [semantic versioning](https://semver.org/). We are currently still in the development release (0.x), which in principle is a mark that the public API is not yet stable. Regardless, we try to maintain semantic versioning also for the development releases. We do so as follows. Any released minor version (0.x.z) will be API backwards-compatible with any previous release of the *same* minor version (0.x.y, with y < z). This means that *if* there is an API incompatible change, we will increase the minor version. For example, release 0.8.1 is API backwards-compatible with release 0.8.0, while release 0.9.0 is API incompatible with version 0.8.1. Note that this only concerns the *public* API, internal functions may change also within a minor version.
```

Comment:
The Branching section is much more detailed than the rest is not directly relevant to contributing. While this is good information, does it belong into this document?  Should we perhaps significantly shorten it and write it in the form of easy-to-follow instructions about whether to target `master` or `develop`?

Reply:
I decided to keep this for the time being so I can merge the PR; we can always re-write it or simplify it in a later step.